### PR TITLE
Fixed the unused parameter in assignProduct

### PR DESCRIPTION
**Pull request by rubanooo**
The defined parameter is never passed to the intended method.

### CODE REVIEW

**Comment by tmotyl** [comment]
@rubanooo can you please describe an issue this patch is solving, so its easier to test the patch.

**Comment by rubanooo** [comment]
> @rubanooo can you please describe an issue this patch is solving, so its easier to test the patch.

The `$identifierType` parameter was not used in the function `assignProduct`, the `$identifierType` should be passed to `_getProductId` so you can assign products to categories based on sku.

**Comment by tmotyl** [comment]
@rubanooo can you outline steps needed to reproduce the issue? Just by reading, the change looks legit, but in order to merge it, it needs to be tested.

**Comment by rubanooo** [comment]
@tmotyl In my case I was writing a custom method to assign categories to products. I found when I tried to sync a product that has a numeric SKU, that the `assignProduct` method would fail because when the `_getProductId` method is ran with a numeric ID, the method parses it as a product ID instead of a product SKU. 

You can test `assignProduct` with a numeric SKU. In this case the method will fail because no product can be found with an ID of the SKU value.

**Comment by Sdfendor** [comment]
Could you add the information that "sku" is the expected value of identifier type (in case not null)? As I understand the called helper within the method, no other value is possible (except the default null).
Besides that good fix.

**Comment by rubanooo** [comment]
@Sdfendor I have added some explanation in the phpdoc

**Comment by Sdfendor** [comment]
Thanks 👍.

**Review by Sdfendor — DISMISSED** [review]

**Review by sreichel — DISMISSED** [review]

File: app/code/core/Mage/Catalog/Model/Category/Api.php [hunk 1]
```diff
@@ -464,7 +467,7 @@ public function assignProduct($categoryId, $productId, $position = null, $identi
     {
         $category = $this->_initCategory($categoryId);
         $positions = $category->getProductsPosition();
-        $productId = $this->_getProductId($productId);
+        $productId = $this->_getProductId($productId, $identifierType);
```
Comment (kiatng) [inline app/code/core/Mage/Catalog/Model/Category/Api.php:470]: Can you fix the docblock for `$identifierType` at line 462?

```
// from
* @param null $identifierType
// to
* @param null|string $identifierType Should be 'sku' when working with SKU's. Leave null when using ID's
```

Comment (rubanooo) [reply app/code/core/Mage/Catalog/Model/Category/Api.php:471]: Added 😃

**Review by kiatng — APPROVED** [review]

**Review by Sdfendor — APPROVED** [review]

**Review by Sekiphp — APPROVED** [review]

**Review by elidrissidev — APPROVED** [review]
Tested and it works fine. To reproduce:
```php
$client = new SoapClient('<magento-host>/api/v2_soap/?wsdl');
$session = $client->login('<username>', '<password>');
$client->catalogCategoryAssignProduct($session, <category-id>, '1209', null, 'sku');
```
Assuming you have a product with numeric sku like `1209`, this throws a `SoapFault` with message `Product not exists.` before this change.

**Review by fballiano — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**